testutil/runner.go

```diff
@@ -399,6 +403,30 @@ func (r *Runner) SendListMessage(channel model.Snowflake) {
 	r.AssertState()
 }
 
+// SendKarmaListMessage senda a ?karmalist message to the bot
+func (r *Runner) SendKarmaListMessage(channel model.Snowflake) {
+	r.T.Helper()
+
+	sendMessage(r.DiscordSession, r.Handler, channel, "?karmalist")
+	r.DiscordMessagesCount++
+	r.GistsCount++
+	assertNewMessages(r.T, r.DiscordSession, []*Message{NewMessage(channel.Format(), "The list of karma is here: https://www.example.com/success")})
+	if r.GistsCount > 0 {
```

Comment:
The goal with the `runner` is to make it so that it asserts the bot's reactions no matter how you modify the bot. So we need to find a way to do that

Let's avoid mimicking my `list` feature tests here. It's basically hard-coded twice and has been a little painful (most recently, I see you found that and it was frustrating. Sorry!)

How about this:
 - Refactor a new file, `feature/karmalist/modelhelper.go`. You can use `feature/vote/modelhelper.go` for inspiration
 - Pull out a function that takes a karmaMap and returns the gist output
 - Use it in the karmalist executor
 - Replace the buffer code below with a call to that with the `KarmaMap` on the `runner` type
 - Add tests for feature/karmalist/modelhelper.go. You can look at the tests for `feature/vote/modelhelper.go` for inspiration. I think you should test (a) empty, (b) two elements with different counts to test sorting, (c) a negative value sorts higher than a positive one
 - Consider adding more tests to the feature_test, but the one you added is good enough IMO when paired with the modelhelper test. They'll split duties: the modelhelper test will ensure that all cases are handled, and the runner test will ensure that all of the different argument configurations work (and there's only one argument configuration, so you only need one test).